sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReduceFnRunner.java

```diff
@@ -837,15 +860,11 @@ public void output(OutputT toOutput) {
   private Instant scheduleEndOfWindowOrGarbageCollectionTimer(
       ReduceFn<?, ?, ?, W>.Context directContext) {
     Instant inputWM = timerInternals.currentInputWatermarkTime();
-    Instant endOfWindow = directContext.window().maxTimestamp();
-    Instant fireTime;
-    String which;
-    if (endOfWindow.isBefore(inputWM)) {
-      fireTime = endOfWindow.plus(windowingStrategy.getAllowedLateness());
+    Instant fireTime = directContext.window().maxTimestamp();
```

Comment:
"fire" is a bit of a loaded word, since triggers share it (whether or not they should is another issue).